*** CMakeLists.txt.orig	2021-11-03 17:21:43.775655459 +0000
--- CMakeLists.txt	2021-11-03 17:24:30.516306583 +0000
*************** option (NPM "Generate NPM/GYP tarballs"
*** 16,21 ****
--- 16,27 ----
  option (BUILDTESTS "Generate check-ups for upm" OFF)
  option (WERROR "Make all warnings into errors." ON)
  
+ 
+ # PYNQ v2.7 - add CMAKE flags to find mraa, clear warning messages
+ set(ENV{PKG_CONFIG_PATH} "/opt/mraa/lib/pkgconfig")
+ cmake_policy(SET CMP0078 OLD)
+ cmake_policy(SET CMP0086 OLD)
+ 
  # Warn if building in source root
  if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
    message (WARNING "Building into sources dir can be risky, prefer other directory")
