From 0e457444d0e102d4fd95d23ee430c441e25f3126 Mon Sep 17 00:00:00 2001
From: Fred Kellerman <fredkellerman@users.noreply.github.com>
Date: Sat, 18 May 2019 23:21:35 -0400
Subject: [PATCH] fix up pwrseq_simple for U96 v2

---
 drivers/mmc/core/pwrseq_simple.c | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/drivers/mmc/core/pwrseq_simple.c b/drivers/mmc/core/pwrseq_simple.c
index 13ef162..59bb337 100644
--- a/drivers/mmc/core/pwrseq_simple.c
+++ b/drivers/mmc/core/pwrseq_simple.c
@@ -105,6 +105,33 @@ static int mmc_pwrseq_simple_probe(struct platform_device *pdev)
 {
 	struct mmc_pwrseq_simple *pwrseq;
 	struct device *dev = &pdev->dev;
+	struct gpio_desc *gpio_chip_en;
+	struct gpio_desc *gpio_resetn;
+
+	// Leave resetn low and let pwrseq take care of it
+	gpio_resetn = gpiod_get(dev, "reset", GPIOD_ASIS);
+	if (IS_ERR(gpio_resetn)) {
+		dev_warn(dev,
+                   "mmc failed to get default resetn GPIO\n");
+	} else {
+		dev_info(dev, "mmc succesfully got gpio_resetn\n");
+		gpiod_direction_output(gpio_resetn, 1);  // low
+		gpiod_put(gpio_resetn);
+	}
+
+	// Turn power off then back on
+	gpio_chip_en = gpiod_get(dev, "chip_en", GPIOD_ASIS);
+	if (IS_ERR(gpio_chip_en)) {
+		dev_warn(dev,
+                   "mmc failed to get default chip_en GPIO\n");
+	} else {
+		dev_info(dev, "mmc succesfully got gpio_chip_en\n");
+		gpiod_direction_output(gpio_chip_en, 1);  // low (chip off)
+		mdelay(5);
+		gpiod_set_value(gpio_chip_en, 0); // high (chip on)
+		gpiod_put(gpio_chip_en);
+		mdelay(5);
+	}
 
 	pwrseq = devm_kzalloc(dev, sizeof(*pwrseq), GFP_KERNEL);
 	if (!pwrseq)
-- 
2.7.4

